include/RAJA/policy/cuda/reduce.hpp

```diff
@@ -62,29 +62,102 @@
 
 #include "RAJA/util/types.hpp"
 
+#include "RAJA/util/basic_mempool.hpp"
+
 #include "RAJA/pattern/reduce.hpp"
 
 #include "RAJA/policy/cuda/MemUtils_CUDA.hpp"
+
 #include "RAJA/policy/cuda/policy.hpp"
+
 #include "RAJA/policy/cuda/raja_cudaerrchk.hpp"
 
 #include <cuda.h>
 
+#include "RAJA/util/mutex.hpp"
+
 namespace RAJA
 {
 
-// HIDDEN namespace to encapsulate helper functions
-namespace HIDDEN
+namespace reduce
+{
+
+namespace cuda
+{
+//! atomic operator version of Reducer object
+template <typename Reducer>
+struct atomic;
+
+template <typename T>
+struct atomic<sum<T>>
+{
+  RAJA_DEVICE RAJA_INLINE
+  void operator()(T &val, const T v)
+  {
+    RAJA::_atomicAdd(&val, v);
```

Comment:
if _atomicAdd is only viable for cuda, shouldn't it live under the `RAJA::cuda` namespace as well?